actionview/lib/action_view/helpers/form_options_helper.rb

```diff
@@ -644,6 +644,24 @@ def time_zone_options_for_select(selected = nil, priority_zones = nil, model = :
       #   collection_radio_buttons(:post, :author_id, Author.all, :id, :name_with_initial) do |b|
       #      b.label(:"data-value" => b.value) { b.radio_button + b.text }
       #   end
+      #
+      # ==== Gotcha
+      #
+      # The HTML specification says when nothing is select on a collection of radio buttons
+      # web browsers do not send any value to server.
+      # Unfortunately this introduces a gotcha:
+      # if a +User+ model has a +category_ids+ field, and in the form none category is selected no +category_ids+ parameter is sent. So,
+      # any mass-assignment idiom like
+      #
+      #   @user.update(params[:user])
+      #
+      # wouldn't update the +category_ids+.
+      #
+      # To prevent this the helper generates an auxiliary hidden field before
+      # every collection of radio buttons. The hidden field has the same name as collection radio button and blank value.
+      #
+      # In case if you don't want the helper to generate this hidden field you can specify
+      # <tt>include_hidden: false</tt> option.
```

Comment:
Hello @maurogeorge – I think this comment is incorrect.

In the case of a collection of radio button, you don't have `category_ids` (plural) and mass assignments.

Instead, the effect of this commit is to tackle the case where you have a form that contains *only* a collection of radio buttons (and no other field type).

For instance, if you have a User#update form, where all you can do is select the category with radio buttons.

If you submit such a form without selecting any radio button, then you expect the params to contain `{user: {category_id: nil}`. This lets you save the object or optionally raise a validation error for the absence of category.

However, the current behavior is that the params will not contain `{user: ...}` at all. 
Therefore if your controller requires strong parameters like `params.require(:user).permit(...)`, the `require` part will fail and the user will see a 400 error.